src/api/lib/rabbitmq_bus.rb

```diff
@@ -1,29 +1,42 @@
 class RabbitmqBus
+  cattr_accessor :connection, :exchange
+
   def self.send_to_bus(channel, data)
     publish("#{Configuration.amqp_namespace}.#{channel}", data)
   rescue => e
     Rails.logger.error "Publishing to RabbitMQ failed: #{e.message}"
     Airbrake.notify(e)
+    # reconnect on next message
+    RabbitmqBus.connection.stop
+    RabbitmqBus.exchange = nil
   end
 
   def self.publish(event_routing_key, event_payload)
     return unless CONFIG['amqp_options']
-    start_connection
+    start_connection unless connection && connection.open?
+    wrapped_exchange.publish(event_payload, routing_key: event_routing_key)
+  end
 
-    $rabbitmq_exchange.publish(event_payload, routing_key: event_routing_key)
+  def self.wrapped_exchange
+    connection.start
+    rabbitmq_channel = connection.create_channel
+    RabbitmqBus.exchange = if CONFIG['amqp_exchange_name']
```

Comment:
:thought_balloon: I don't like this style of assignment. Makes it less clear what is happening...

Reply:
I only moved this. I guess next time you pass by a mirror, you discuss this with the author of
https://github.com/bgeuken/open-build-service/blob/4d57860fb0ff6b37ac61f9d2c425b7f0776ee650/src/api/app/models/rabbitmq_bus.rb